priam/src/main/java/com/netflix/priam/defaultimpl/CassandraProcessManager.java

```diff
@@ -154,6 +157,34 @@ public void stop() throws IOException {
         stopCass.redirectErrorStream(true);
 
         instanceState.setShouldCassandraBeAlive(false);
+        if (config.getGracefulDrainHealthWaitSeconds() > 0) {
+            ExecutorService executor = Executors.newSingleThreadExecutor();
+            Future drainFuture = executor.submit(() -> {
+                // As the node has been marked as shutting down above in setShouldCassandraBeAlive, we wait this
+                // duration to allow external healthcheck systems time to pick up the state change.
+                try {
+                    Thread.sleep(config.getGracefulDrainHealthWaitSeconds() * 1000);
+                } catch (InterruptedException e) {
+                    return;
+                }
+
+                try {
+                    JMXNodeTool nodetool = JMXNodeTool.instance(config);
+                    nodetool.drain();
+                } catch (InterruptedException | IOException | ExecutionException e) {
+                    logger.error("Exception draining Cassandra, could not drain. Proceeding with shutdown.", e);
+                }
+                // Once Cassandra is drained the thrift/native servers are shutdown and there is no need to wait to
+                // stop Cassandra. Just stop it now.
+            });
+
+            // In case drain hangs, timeout the future and continue stopping anyways.
+            try {
+                drainFuture.get(2 * config.getGracefulDrainHealthWaitSeconds(), TimeUnit.SECONDS);
+            } catch (ExecutionException | TimeoutException | InterruptedException e) {
```

Comment:
Do you want to use different error here, stating that waited config amount of time for the drain to complete, but proceeding? so that we can identify pattern if any

Reply:
And a different timeout config/ constant. So that we give enough time for drain always regardless of getGracefulDrainHealthWaitSeconds setting

Reply:
Yes, I can fix that up.